operators/project.py

```diff
@@ -119,71 +128,93 @@ def draw(self, context):
-            nonlocal texture
-            if hasattr(gen, '_active_generation_future'):
-                del gen._active_generation_future
-            context.scene.dream_textures_info = ""
-            context.scene.dream_textures_progress = 0
-            generated = future.result()
-            if isinstance(generated, list):
-                generated = generated[-1]
-            if texture is None:
-                texture = bpy.data.images.new(name=str(generated.seed), width=generated.image.shape[1], height=generated.image.shape[0])
-            texture.name = str(generated.seed)
-            material.name = str(generated.seed)
-            texture.pixels[:] = generated.image.ravel()
-            texture.update()
-            image_texture_node.image = texture
-        
-        def on_exception(_, exception):
-            context.scene.dream_textures_info = ""
-            context.scene.dream_textures_progress = 0
-            if hasattr(gen, '_active_generation_future'):
-                del gen._active_generation_future
-            raise exception
-        
-        context.scene.dream_textures_info = "Starting..."
-        future = gen.depth_to_image(
-            depth=depth,
-            image=init_img_path,
-            **context.scene.dream_textures_project_prompt.generate_args()
-        )
-        gen._active_generation_future = future
-        future.call_done_on_exception = False
-        future.add_response_callback(on_response)
-        future.add_done_callback(on_done)
-        future.add_exception_callback(on_exception)
+        shader_info.vertex_source("""
+void main()
+{
+    gl_Position = vec4(dest_uv * 2 - 1, 0.0, 1.0);
+    uvInterp = src_uv;
+}
+""")
+
+        shader_info.fragment_source("""
+void main()
+{
+    fragColor = texture(image, uvInterp);
+}
+""")
 
-    finally:
-        cleanup()
+        return gpu.shader.create_from_info(shader_info)
+
+    width, height = dest.size[0], dest.size[1]
+    offscreen = gpu.types.GPUOffScreen(width, height)
+
+    buffer = gpu.types.Buffer('FLOAT', width * height * 4, src)
+    texture = gpu.types.GPUTexture(size=(width, height), data=buffer, format='RGBA16F')
+
+    with offscreen.bind():
+        fb = gpu.state.active_framebuffer_get()
+        fb.clear(color=(0.0, 0.0, 0.0, 0.0))
+        gpu.state.depth_test_set('LESS_EQUAL')
+        gpu.state.depth_mask_set(True)
+        with gpu.matrix.push_pop():
+            gpu.matrix.load_matrix(mathutils.Matrix.Identity(4))
+            gpu.matrix.load_projection_matrix(mathutils.Matrix.Identity(4))
+
+            vertices = np.array([[l.vert.index for l in loop] for loop in mesh.calc_loop_triangles()], dtype='i')
+
+            shader = bake_shader()
+            batch = batch_for_shader(
+                shader, 'TRIS',
+                {"src_uv": src_uv, "dest_uv": dest_uv},
+                indices=vertices,
+            )
+            shader.uniform_sampler("image", texture)
+            batch.draw(shader)
```

Comment:
'LESS_EQUAL' and 'LESS' depth_test modes don't appear to work at all on my system, and none of the rest are getting any better result than coloring all of the selected UV zones a single color.

The viewport freezes and other spaces had issues redrawing when depth_test is left as 'LESS_EQUAL'
```suggestion
        prev_depth_test = gpu.state.depth_test_get()
        prev_depth_mask = gpu.state.depth_mask_get()
        gpu.state.depth_test_set('LESS_EQUAL')
        gpu.state.depth_mask_set(True)
        with gpu.matrix.push_pop():
            gpu.matrix.load_matrix(mathutils.Matrix.Identity(4))
            gpu.matrix.load_projection_matrix(mathutils.Matrix.Identity(4))

            vertices = np.array([[l.vert.index for l in loop] for loop in mesh.calc_loop_triangles()], dtype='i')

            shader = bake_shader()
            batch = batch_for_shader(
                shader, 'TRIS',
                {"src_uv": src_uv, "dest_uv": dest_uv},
                indices=vertices,
            )
            shader.uniform_sampler("image", texture)
            batch.draw(shader)
        gpu.state.depth_test_set(prev_depth_test)
        gpu.state.depth_mask_set(prev_depth_mask)
```